src/telemetry/logs.ts

```diff
@@ -201,92 +277,56 @@ export class DBOSContextualLogger implements DLogger {
-    readonly telemetryCollector: TelemetryCollector,
-    logLevel: string,
-  ) {
-    super();
-    this.level = logLevel;
-    // not sure if we need a more explicit name here
-    const { LoggerProvider } = require('@opentelemetry/sdk-logs');
-    const loggerProvider = new LoggerProvider();
-    this.otelLogger = loggerProvider.getLogger('default');
-    this.applicationID = globalParams.appID;
-    this.executorID = globalParams.executorID;
-    const logRecordProcessor = {
-      forceFlush: async () => {
-        // no-op
-      },
-      onEmit(logRecord: LogRecord) {
-        telemetryCollector.push(logRecord);
-      },
-      shutdown: async () => {
-        // no-op
-      },
-    };
-    loggerProvider.addLogRecordProcessor(logRecordProcessor);
+  debug(logEntry: unknown, _metadata?: ContextualMetadata): void {
+    if (this.config.logLevel === 'debug') {
+      console.debug(logEntry);
+    }
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  log(info: any, callback: () => void): void {
-    const { level, message, stack, span } = info;
-
-    const levelToSeverityNumber: { [key: string]: SeverityNumber } = {
-      error: SeverityNumber.ERROR,
-      warn: SeverityNumber.WARN,
-      info: SeverityNumber.INFO,
-      debug: SeverityNumber.DEBUG,
-    };
-
-    // Ideally we want to give the spanContext to the logRecord,
-    // But there seems to some dependency bugs in opentelemetry-js
-    // (span.getValue(SPAN_KEY) undefined when we pass the context, as commented bellow)
-    // So for now we get the traceId and spanId directly from the context and pass them through the logRecord attributes
-    this.otelLogger.emit({
-      severityNumber: levelToSeverityNumber[level as string],
-      severityText: level as string,
-      body: message as string,
-      timestamp: performance.now(), // So far I don't see a major difference between this and observedTimestamp
-      observedTimestamp: performance.now(),
-      attributes: {
-        ...span?.attributes,
-        traceId: span?.spanContext()?.traceId,
-        spanId: span?.spanContext()?.spanId,
-        stack,
-        applicationID: this.applicationID,
-        applicationVersion: globalParams.appVersion,
-        executorID: this.executorID,
-      } as LogAttributes,
-    });
+  warn(logEntry: unknown, _metadata?: ContextualMetadata): void {
+    console.warn(logEntry);
+  }
 
-    callback();
+  error(inputError: unknown, metadata?: ContextualMetadata & StackTrace): void {
+    if (inputError instanceof Error) {
+      console.error(inputError);
+    } else if (metadata?.stack) {
+      console.error(inputError, '\n', metadata.stack);
+    } else {
+      console.error(inputError);
+    }
   }
 }
+
+/***********************/
+/* FORMAT & TRANSPORTS */
+/***********************/
+
+// Lazily create the console format only when Winston is used
+function getConsoleFormat() {
+  const { format } = require('winston') as { format: typeof formatT };
```

Comment:
Do we want a try-catch here to avoid crashing the entire process?

Reply:
No--if you enable OTLP but don't have the OTLP dependencies, you should get a hard error.